### DISCUSSION TITLE
Foreign key constraint failed on the field: `foreign key`

### DISCUSSION THREAD

**Discussion by mufarodev** [prisma/prisma#17431] [Q&A]
Hi,

I'm trying to delete the guild row, but I'm getting an error about foreign key. I've been trying to do a research on this, but couldn't get to understand it. Can someone give me an advice on what should I change here?

schema.prisma
```prisma
model Message {
  id        Int      @id @default(autoincrement())
  content   String
  userId    Int
  createdAt DateTime @default(now())
  user      User     @relation(fields: [userId], references: [id])
  channelId Int      @default(0)
  channel   Channel  @relation(fields: [channelId], references: [id])
}

model Guild {
  id        Int           @id @default(autoincrement())
  name      String
  icon      String
  ownerId   Int
  createdAt DateTime      @default(now())
  channels  Channel[]
  members   GuildMember[]
  invites   Invite[]
}

model Channel {
  id        Int       @id @default(autoincrement())
  name      String
  guildId   Int
  createdAt DateTime  @default(now())
  guild     Guild     @relation(fields: [guildId], references: [id])
  messages  Message[]
}

model GuildMember {
  id        Int      @id @default(autoincrement())
  userId    Int
  guildId   Int
  createdAt DateTime @default(now())
  user      User     @relation(fields: [userId], references: [id])
  guild     Guild    @relation(fields: [guildId], references: [id])
}

model Invite {
  id        Int      @id @default(autoincrement())
  code      String   @unique
  createdAt DateTime @default(now())
  guildId   Int
  guild     Guild    @relation(fields: [guildId], references: [id])
}

```

Query
```typescript
await prisma.guild.delete({
  where: {
    id: guildId,
  },
 });
```

**Comment by nurul3101** [comment]
Hello @xMufaro 👋 

From the schema file, I can see that there are no [referential actions](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/referential-actions) set up. You can set them up to automatically deleted connecting records first before deleting the parent record.

You might receive this error when you are deleting a guild record which has connected channels, GuildMember, or Invites.
You can try deleting these connected records first before deleting the Guild.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
